esmvaltool/utils/cmorizers/obs/utilities.py

```diff
@@ -191,14 +191,15 @@ def set_global_atts(cube, attrs):
             'comment':
             attrs.pop('comment'),
             'user':
-            os.environ["USER"],
+            os.environ.get("USER"),
```

Comment:
Maybe it would be a bit simpler to use
```suggestion
            os.environ.get("USER", "None"),
```
and remove the `_dict_clean` function?

Reply:
I did not now about that option, neat. 
But _dict_clean would also handle all other "None"s for making sure we get only strings, just in case.

It should not fail because of that.

Reply:
I think it's a bit confusing to not create the dict correctly in the first place, but correct issues later on in another function.

To make sure that everything's a string, it's probably better to do something like:

`glob_dict = {key: str(value) for key, value in glob_dict.items()}`

because now you're only correcting `None` objects.

Reply:
Most attributes are defined outside the utilities = Other users' scripts.

I included the _dict_clean for having an option to fix known issues; especially, those who are not caught and warned for by the except statement. 
If you want to string all values, I'm fine with it. I'm just not sure why we should avoid/**not** have a method/an option for adapting to known issues (like we do with all the fixes).

Reply:
Automatically fixing the scripts that should do the fixing will lead to poor code quality of those scripts, that's why I'm not too keen on it, but I leave it up to you.

I looked at some model data and apparently numerical values are fine, so converting everything to strings is probably not needed.

Reply:
@schlunma @valeriupredoi any preferences?

Reply:
I don't think we need `_dict_clean`: I don't see any case where one of the dictionary values is `None` except the user sets them to `null` in the configuration file. So adding something like this

```python
os.environ.get("USER", 'unknown user')
os.environ.get("HOSTNAME", 'unknown host')
```

should be enough.

Reply:
Ok, but don't blame me, if we need more than that at one point ;)

It's the best solution for the current needs, in deed.